guardian/bonobo
# Updating Dynamo search function to find users with no keys

## What does this change?
This allows searching for users that have no keys in Bonobo, and consequently adding keys to those users. 

Why was this done?
Bonobo search used to only bring up users who have keys, which meant if a users only key was deleted, there was no way to recreate their key in the UI. 

## How to test

Locally:
-Run Bonobo locally following the steps in the readme file, make sure the application is pointed to CODE DynamoDB tables (in the application.conf file)

On CODE:
-Go to bonobo-code.capu.gutools.co.uk

-Create a user and a key
-Delete that user's key
-Search for user using their name, email, or company name
-A record with that user's details should show up (and the key field should appear empty). You should be able to click on that user record and add/delete keys as you wish. If a user has no keys, the Created at field would display the user's creation date rather than the key's creation date


## How can we measure success?
Better user experience when recreating keys that have been deleted.

## Have we considered potential risks?
This is a low risk change.
